qiskit/simulators/_qasm_cpp_simulator.py

```diff
@@ -73,28 +103,48 @@ def __init__(self, job):
                 cmd = '"{0}" or "{1}" '.format(self._exe, './' + self._exe)
                 raise FileNotFoundError(cmd)
 
-    def run(self, silent=True):
+    def run(self):
         """
         Run simulation on C++ simulator.
         """
+        result_list = []
+        for circuit in self.qobj['circuits']:
+            result_list.append( self.run_circuit(circuit) )
+        return Result(result_list, self.qobj)
```

Comment:
In the same spirit as the comment at `qiskit/_result.py:71`, a `from qiskit import Result` seems to be missing.

Reply:
Also, would it be possible to double-check that this simulator is returning the right value? I'm unsure if it is by design, but the rest of simulators' `run()` method seem to be returning:

```python
return Result({'result': result_list, 'status': 'COMPLETED'},
              self.qobj)
```

Reply:
Good catch! We should probably also have tests for the c++ simulator to catch this kind of thing.